pyproject.toml

```diff
@@ -53,22 +52,21 @@ dependencies = [
     "boltons >=23.0.0",
     "charset-normalizer",
     "zstandard >=0.15",
-    # The following dependencies were added in the process of vendoring Poetry 1.8.3.
-    "build >=1.0.3,<2.0.0",
+    # The following dependencies were added in the process of vendoring Poetry 2.0.1.
+    "build >=1.2.1,<2.0.0",
     "cachecontrol[filecache] >=0.14.0,<0.15.0",
     "crashtest >=0.4.1,<0.5.0",
-    "dulwich >=0.21.2,<0.22.0",
+    "dulwich >=0.22.6,<0.23.0",
     "fastjsonschema >=2.18.0,<3.0.0",
     'importlib-metadata >= 4.4 ;  python_version <"3.10"',
     "installer >=0.7.0,<0.8.0",
-    "keyring >=24.0.0,<25.0.0",
-    "pexpect >=4.7.0,<5.0.0",
-    "pkginfo >=1.10,<2.0",
+    "keyring >=25.1.0,<26.0.0",
+    "pkginfo >=1.12,<2.0",
     "pyproject-hooks >=1.0.0,<2.0.0",
     "requests-toolbelt >=1.0.0,<2.0.0",
     "shellingham >=1.5,<2.0",
     "trove-classifiers >= 2022.5.19",
-    "virtualenv >=20.23.0,<21.0.0",
+    "virtualenv >=20.26.6,<21.0.0",
     'xattr >=1.0.0,<2.0.0 ; sys_platform == "darwin"',
 ]
 
```

Comment:
I double-checked the dependencies, and I get the same result.

In the process I realized that I made things way more complicated than they need to be by deduplicating the dependencies.

It's fine to include duplicates, and the constraints will automatically be merged. So the much simpler way to do this is to straight-up copy-paste the deps from the vendored conda, python, and cleo without modification (but of course add appropriate comments to indicate the respective blocks).

No need to do this now, but for next time.